src/devices/LedDisplay/SevenSegmentFontHelper.cs

```diff
@@ -0,0 +1,162 @@
+            /* 1 */ Segment.TopRight | Segment.BottomRight,
+            /* 2 */ Segment.Top | Segment.TopRight | Segment.Bottom | Segment.BottomLeft | Segment.Middle,
+            /* 3 */ Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.Middle,
+            /* 4 */ Segment.TopRight | Segment.BottomRight | Segment.TopLeft | Segment.Middle,
+            /* 5 */ Segment.Top | Segment.BottomRight | Segment.Bottom | Segment.TopLeft | Segment.Middle,
+            /* 6 */ Segment.Top | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft | Segment.Middle,
+            /* 7 */ Segment.Top | Segment.TopRight | Segment.BottomRight,
+            /* 8 */ Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft | Segment.Middle,
+            /* 9 */ Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.TopLeft | Segment.Middle,
+            /* a */ Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.BottomLeft | Segment.TopLeft | Segment.Middle,
+            /* b */ Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft | Segment.Middle,
+            /* C */ Segment.Top | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft,
+            /* d */ Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.Middle,
+            /* E */ Segment.Top | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft | Segment.Middle,
+            /* F */ Segment.Top | Segment.BottomLeft | Segment.TopLeft | Segment.Middle
+        };
+
+        /// <summary>
+        /// Convert byte value hexadecimal digit (0..F) to corresponding font bits
+        /// </summary>
+        /// <param name="digit">hexadecimal digit (0..F)</param>
+        /// <returns>corresponding font segments</returns>
+        public static Segment GetHexDigit(byte digit) => s_hexDigitFonts[digit & HexadecimalMask];
+
+        /// <summary>
+        /// Return a list of hexadecimal digits to their segmented font representation
+        /// </summary>
+        /// <param name="digits">list of hexadecimal digits</param>
+        /// <returns>list of corresponding font segments</returns>
+        public static ReadOnlySpan<Segment> GetHexDigits(ReadOnlySpan<byte> digits)
+        {
+            var hexDigits = new Span<Segment>(new Segment[digits.Length]);
+
+            int i = 0;
+            foreach (var digit in digits)
+            {
+                hexDigits[i++] = GetHexDigit(digit);
+            }
+
+            return hexDigits;
+        }
+
+        /// <summary>
+        /// Character font table
+        /// </summary>
+        private static readonly IReadOnlyDictionary<char, Segment> s_characterFontTable = new Dictionary<char, Segment>
+        {
+            { '0', Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft },
```

Comment:
I really prefer the format the other binding has:

```csharp
[Flags]
enum Character
{
    TopSegment,
    BottomSegment,
    ...
    Zero = Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft
}
```

then this whole dictionary would be also a bit clearer:
`{ '0', Character.Zero },`

etc.

Also other thing is I think this could be just something like:

```csharp
static Character[] s_digits = { Character.Zero, Character.One, ... };
static Character[] s_lowerCaseLetters = { Character.SmallLetterA };
...
```

and just right math and few ifs should give you the right output but I'm fine with dictionary as well.